src/actions/tamanu.rs

```diff
@@ -53,3 +60,40 @@ pub fn find_tamanu(args: &TamanuArgs) -> Result<(Version, PathBuf)> {
 			.ok_or_else(|| miette!("no tamanu discovered, use --root"))
 	}
 }
+
+pub fn find_package(root: impl AsRef<Path>) -> Result<String> {
+	fs::read_dir(root.as_ref().join("packages"))
+		.into_diagnostic()?
+		.filter_map_ok(|e| e.file_name().into_string().ok())
+		.process_results(|mut iter| {
+			iter.find(|dir_name| dir_name == "central-server" || dir_name == "facility-server")
+				.ok_or_else(|| miette!("Tamanu servers not found"))
+		})
+		.into_diagnostic()?
+}
+
+pub fn find_existing_version() -> Result<Version> {
```

Comment:
This won't work at all on non-Windows, so the way to go would be to _both_:

- mark this `#[cfg(windows)]`
- mark the entire `pre-upgrade` command `#[cfg(windows)]`